python/mlx/nn/layers/base.py

```diff
@@ -90,6 +91,34 @@ def __getattr__(self, key: str):
     def __setattr__(self, key: str, val: Any):
         self[key] = val
 
+    def __deepcopy__(self, memo):
+        cls = self.__class__
+        instance_copy = cls.__new__(cls)
+        memo[id(self)] = instance_copy
+
+        for k, v in self.items():
+            if isinstance(v, mx.array):
+                obj = mx.array(v)
+            elif isinstance(v, list):
+                obj = [
+                    mx.array(elem) if isinstance(elem, mx.array) else deepcopy(elem)
```

Comment:
This solution might not be the best but I did not find a method that copies `mx.array` so I just copied it by calling `mx.array(old_mx_array)` to create a new array.